supersid/supersid.py

```diff
@@ -132,43 +161,51 @@ def on_timer(self):
         self.viewer.status_display(message, level=1)
         signal_strengths = []
         try:
-            data = self.sampler.capture_1sec()  # return list of signal strength
-            Pxx, freqs = self.psd(data, self.sampler.NFFT,
-                                  self.sampler.audio_sampling_rate)
-            for binSample in self.sampler.monitored_bins:
-                signal_strengths.append(Pxx[binSample])
+            # capture_1sec() returns list of signal strength,
+            # may set sampler_ok = False
+            data = self.sampler.capture_1sec()
+
+            if self.sampler.sampler_ok:
+                Pxx, freqs = self.psd(data, self.sampler.NFFT,
+                                      self.sampler.audio_sampling_rate)
+                for binSample in self.sampler.monitored_bins:
+                    signal_strengths.append(Pxx[binSample])
         except IndexError as idxerr:
             print("Index Error:", idxerr)
             print("Data len:", len(data))
         except TypeError as err_te:
             print("Warning:", err_te)
 
-        # ensure that one thread at the time accesses the sid_file's' buffers
-        with self.timer.lock:
```

Comment:
I see you removed the lock. Is there still a mechanism to make sure we only have 1 thread in the on_timer() code at a time?

Reply:
No I did not remove the lock, I moved it one level up into sidtimer.py, class SidTimer(), method _ontimer().
This way not just the file access is locked but also capturing audio data from the sound card.

Reply:
Great. Thanks